### PR TITLE
generating ci config

### PR DESCRIPTION
**Pull request by deanshub**
### 🔦 Summary
generating .ci_config file

### CODE REVIEW

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import path from 'path';
+import fs from 'fs-extra';
+
+export function generateCiConfig(
+  appDefId: string,
+  widgetName: string,
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: appName

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  baseUrl: string,
+  subWidgets: Array<{ id: string; name: string }>,
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: Widgets

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+) {
+  const ciConfig = {
+    app_def_id: appDefId,
+    ignore_dependencies: 'clear',
+    tpa_url_templates: [
+      `appFields.platform.baseUrls.${widgetName}BaseUrl=>https://static.parastorage.com/services/${baseUrl}/{version}/`,
+      "widgets[?(@.widgetId=='13afb094-84f9-739f-44fd-78d036adb028')].componentFields.controllerUrl=>https://static.parastorage.com/services/wixstores-client-gallery/{version}/galleryController.bundle.min.js",
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: Hard coded

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      ...subWidgets.reduce((res: Array<string>, { id, name }) => {
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: widgets

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        return [
+          ...res,
+          `widgets[?(@.widgetId=='${id}')].componentFields.controllerUrl=>https://static.parastorage.com/services/${baseUrl}/{version}/${name}.bundle.min.js`,
+          `widgets[?(@.widgetId=='${id}')].componentFields.controllerUrl=>https://static.parastorage.com/services/${baseUrl}/{version}/${name}Controller.bundle.min.js`,
+        ];
+      }, []),
+    ],
+  };
+  console.log(path.join(process.cwd(), '.ci_config'));
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: non relevant

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  fs.outputFileSync('.ci_config', JSON.stringify(ciConfig, null, 2));
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: fs.outputJson

Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: `.ci_config` -> put in constants

File: packages/yoshi-flow-editor/src/commands/build.ts [hunk 7]
```diff
@@ -74,6 +75,8 @@ const build: cliCommand = async function(argv, config) {
   await copyTemplates();
 
   if (inTeamCity()) {
+    // TODO: make the parameters dynamic
+    generateCiConfig('', 'todo', 'wixstores-client-todo', []);
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/build.ts]: Let's make those parameters dynamic before merging this PR

File: packages/yoshi-flow-editor/src/constants.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export enum FileNames {
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/constants.ts]: Is there a benefit of using `enum` over exporting named properties?

for example:
```ts
export const CI_CONFIG = '.ci_config';
```

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    .concat(
+      widgets.map(widget => generateWidgetControllerUrl(widget, baseUrl)),
+    );
+}
+
+function generateWidgetControllerUrl(
+  { id, name }: Widget,
+  baseUrl: string,
+): string {
+  return `widgets[?(@.widgetId=='${id}')].componentFields.controllerUrl=>https://static.parastorage.com/services/${baseUrl}/{version}/${name}Controller.bundle.min.js`;
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: I'm not sure that that's the case for all apps. Not everybody uses `controllerURL` yet. So I think that it wouldn't change anything for apps that has a single `viewerScript`.

Also, for those kinds of apps, you need to also upgrade their `viewerScript` version.

File: packages/yoshi-flow-editor/src/constants.ts (hunk 8 above)
Comment (deanshub) [reply packages/yoshi-flow-editor/src/constants.ts]: nope, just makes it easier to understand that these are file names

Comment (ranyitz) [reply packages/yoshi-flow-editor/src/constants.ts]: We use this convention to describe file names / directory names - https://github.com/wix/yoshi/blob/master/packages/yoshi-config/src/paths.ts#L16-L19

File: packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts (hunk 9 above)
Comment (deanshub) [reply packages/yoshi-flow-editor/src/commands/ciConfigGenerator.ts]: added both methods just in case

**Comment by deanshub** [comment]
1. create type for the controller config by moving it to native components infra
2. remove empty ctrl (not needed because we throw an error on build)
3. remove platform config file (move them to named exports for appdefid, and support default separate controller pattern)
4. minimize glue code in viewer script wrapping (without controller factory)

**Pull request closed without merging**